### PR TITLE
[next] Fix Vercel creating a Node.js lambda function when using `basePath`,  `edge` runtime, app dir and a top level `page.jsx`

### PR DESCRIPTION
**Pull request by remorses**

### CODE REVIEW

File: packages/next/src/index.ts [hunk 1]
```diff
@@ -2720,7 +2720,18 @@ async function getServerlessPages(params: {
...
     middlewareManifest?.functions ?? {}
   )) {
-    const edgePath = (edgeFunctionFile.slice(1) || 'index') + '.js';
+    let fileName = edgeFunctionFile.slice(1) || 'index';
+    // when using app directory the index page will have path /page
+    if (
+      fileName === 'page' &&
+      // `page` is index path only when using app directory
+      middlewareManifest?.functions?.[fileName + '.js']?.files?.some(
+        (file: string) => file.includes('/app/')
```
Comment (styfle) [inline packages/next/src/index.ts]: I'm not sure this is the right way to detect App Router.

For example, what if a file was named `/components/app/file.js`?

I think you want to look in `appPaths` or `normalizedAppPaths`.

cc @ijjk

Comment (remorses) [reply packages/next/src/index.ts]: I am now using the function name to detect if it’s using app dir

**Comment by styfle** [comment]
I'm seeing a test failure

``` 
@vercel/next:test:   ● 00-app-dir-i18n › should deploy and pass probe checks
@vercel/next:test: 
@vercel/next:test:     State of https://00-app-dir-i18n-qu4e7np9z-vtest314-5jujcc1wreshkiqngpfk.vercel.app/ is ERROR: ENOENT: no such file or directory, open '/vercel/path1/.next/server/pages/en/404.html'
```

https://github.com/vercel/vercel/actions/runs/6027252919/job/16352140467?pr=10401#step:9:1335

**Comment by styfle** [comment]
This might be fixed by https://github.com/vercel/vercel/pull/10416

**Comment by remorses** [comment]
Thank you for that, do you also want me to run changeset? I saw that in other PRs but not in CONTRIBUTING.md

**Comment by styfle** [comment]
Yes changesets is required. I pushed it up and also started CI in https://github.com/vercel/vercel/pull/10463

**Review by styfle — DISMISSED** [review]
This looks like it works once deployed (although typescript was missing from dependencies).

However, I would prefer a e2e test with probes that verifies the runtime behavior (not just build time)

Take a look at https://github.com/vercel/vercel/blob/main/packages/next/test/fixtures/00-edge-runtime/probes.json

**Comment by styfle** [comment]
Thanks for the PR! 

I'm going to close this one in favor of PR https://github.com/vercel/vercel/pull/10465 which is a more complete solution

**Pull request closed without merging**